gspread/worksheet.py

```diff
@@ -2607,6 +2607,28 @@ def unmerge_cells(self, name):
 
         return self.spreadsheet.batch_update(body)
 
+    def get_notes(self):
+        """Returns a list of lists containing all notes in the sheet, or the empty list if the
+        sheet does not have a note.
+        """
+        url = SPREADSHEET_URL % (self.spreadsheet.id)
+        params = {"fields": "sheets/data/rowData/values/note"}
+        response = self.client.request("get", url, params=params)
+        response.raise_for_status()
+        response_json = response.json()
```

Comment:
Please use the dedicated method to obtain the notes: `self.client.spreadsheets_get(self.spreadsheet_id, params)`

it will build the necessary URL for you and return a JSON, as well as raising a dedicated exception in case of error.